pbft/pbft.cpp

```diff
@@ -293,49 +315,49 @@ pbft::common_message_setup(const pbft_operation& op, pbft_msg_type type)
 }
 
 void
-pbft::do_preprepare(pbft_operation& op)
+pbft::do_preprepare(const std::shared_ptr<pbft_operation>& op)
```

Comment:
I'd avoid references to shared pointers unless you are absolutely sure the caller of this will never lose its ref count and it goes to zero.

Reply:
I have been consistent about that. We make a real copy of the pointer in handle_[message type], then everything from there uses references to that copy. I wanted to avoid copying it for all the small functions.

Reply:
If you're not sharing ownership and the original holder of the reference doesn't go out of scope then this is fine.